examples/example_emscripten_wgpu/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+# Example usage:
+#  cmake -B build -g "Ninja"
+
+cmake_minimum_required(VERSION 3.10.2)
+project(imgui_example_emscripten_wgpu C CXX)
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING "" FORCE)
+endif()
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DVK_PROTOTYPES")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DVK_PROTOTYPES")
+
+set(BUILD_TESTING OFF)
+set(BUILD_TESTING_STATIC  OFF)
+set(BUILD_TESTING_SHARED OFF)
+
+# Dear ImGui
+set(IMGUI_DIR ../../)
+include_directories(${IMGUI_DIR} ${IMGUI_DIR}/backends ..)
+
+# Libraries
+if(EMSCRIPTEN)
+	set(LIBRARIES glfw)
+	add_compile_options(-sDISABLE_EXCEPTION_CATCHING=1 -DIMGUI_DISABLE_FILE_FUNCTIONS=1)
+else()
+	# Dawn wgpu desktop
+	set(DAWN_FETCH_DEPENDENCIES ON)
+	add_subdirectory("external/libraries/dawn" dawn EXCLUDE_FROM_ALL) # point this to up to date dawn repo
```

Comment:
It would probably be better to allow users to set DAWN_DIR variable and use that instead so that 

1. The library can be outside of Dear ImGui's source path
2. You could clearly print the error if the variable is not set.

Reply:
I was unsure the best approach to handle this, might need to account for wgpu-native as well (which I had not)